src/Path/CurrentPathStack.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Retrofit\Drupal\Path;
+
+use Drupal\Core\Path\CurrentPathStack as CoreCurrentPathStack;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\RequestStack;
+
+class CurrentPathStack extends CoreCurrentPathStack
+{
+    public function __construct(RequestStack $request_stack)
+    {
+        parent::__construct($request_stack);
+        if ($request = $request_stack->getCurrentRequest()) {
+            $_GET['q'] = $request->getPathInfo();
+            $this->paths[$request] = ['path' => &$_GET['q']];
+        }
+    }
```

Comment:
Should this be called here? I think we just need to set it during `setPath`.

Reply:
How would it be set if `setPath()` is not called?

Reply:
Have there been situations you experienced where it wasn't set? Also, in the constructor, we have a chance of this being constructed before the request is determined.

Reply:
We could mimic `GlobalUserSetter` to set the initial path on `KernelEvents::REQUEST`